CONTRIBUTING.md

```diff
@@ -1,35 +1,88 @@
+- Documentation like what you are currently reading
+- The demo notebooks
+- The code base
 
-Your code should merge cleanly into the codebase; if it does not, you should rebase it, resolve any incompatibilities, and update your PR. Committing to the branch where your PR is committed will do this for you automatically. Note that if your PR is open for too long, merge conflicts will happen as upstream dev changes.
+We welcome all such contributions! To do this you need to make changes on your local version of Caiman then push this to the repository by making a **Pull Request** (PR). We will walk you through this process in the rest of the document. 
+
+Before you go through the work required to improve something, we recommend that you let us know your plans. E.g., open an issue, or reach out on Gitter. This way, we can avoid duplicated effort (if someone is already working on it), or wasted time (it could turn out the changes might not be feasible right now). We can usually set up a video chat on Google or Skype to talk about a feature proposal if that works for you.
+
+## Background: how does this work?
+In this section we'll give general background on how making a contribution/PR works. If you just want to get started quickly, feel free to skip to the next section.
+
+There are many possible workflows for contributing to open source projects (see a summary [here](https://docs.gitlab.com/ee/topics/gitlab_flow.html)). The basic structure of the workflow we use at Caiman is illustrated here:
+
+<img src="docs/img/gitflow.jpg">
+
+When you are working on a git repository, you are always working on a *branch*, and each branch has a name. Some branches are special, `main`, which contains the stable version of the software for general release. Different feature branches, which will have names like `fix_plotting_bug` are usually short-lived and what individual contributors like you will work on. 
+
+To keep a buffer between the main branch and individiaul feature branches, there is a second persistent branch, a development branch called `dev`. This the branch that contributors actively push their changes to, and is the one with all the newest developments.
+
+The workflow for contributing to Caiman is roughly illustrated by the numbers in the above diagram:
+1) Create a feature branch from `dev`.
+2) Work on that feature until you are ready to push it to Caiman.
+3) Make a PR: this is when you request that your changes become merged into `dev` at Caiman. This merge won't be immediate, you will get feedback on your code, and probably be asked to make some changes. 
+4) Eventually, once enough new features accumulate in the `dev` branch to merit a new release (every month or so), the `dev` branch will be merged with `main`. This will become a new version of Caiman that people download when they run `mamba install caiman`. 
```

Comment:
"Eventually, once..." -> "Periodically, as features accumulate in the dev branch, the core developers will merge it into `main` as part of the release process"